src/Compression/CompressionCodecDeflate.cpp

```diff
@@ -0,0 +1,377 @@
+    }
+}
+
+DeflateJobHWPool::~DeflateJobHWPool()
+{
+    for (uint32_t i = 0; i < JOB_POOL_SIZE; ++i)
+    {
+        if (jobPool[i] != nullptr)
+        {
+            while (!tryLockJob(i));
+            qpl_fini_job(jobPool[i]);
+        }
+        jobPool[i] = nullptr;
+        jobLocks[i].store(false);
+    }
+    jobPoolReady() = false;
+}
+
+//HardwareCodecDeflate
+HardwareCodecDeflate::HardwareCodecDeflate():
+    hwEnabled(DeflateJobHWPool::instance().jobPoolReady()),
+    log(&Poco::Logger::get("HardwareCodecDeflate"))
+{
+}
+
+HardwareCodecDeflate::~HardwareCodecDeflate()
+{
+    if (!jobDecompAsyncMap.empty())
+    {
+        LOG_WARNING(log, "Find un-released job when HardwareCodecDeflate destroy");
+        for (auto it : jobDecompAsyncMap)
+        {
+            DeflateJobHWPool::instance().releaseJob(it.first);
+        }
+        jobDecompAsyncMap.clear();
+    }
+}
+uint32_t HardwareCodecDeflate::doCompressData(const char * source, uint32_t source_size, char * dest, uint32_t dest_size) const
+{
+    uint32_t job_id = 0;
+    qpl_job * job_ptr = DeflateJobHWPool::instance().acquireJob(&job_id);
+    if (job_ptr == nullptr)
+    {
+        LOG_WARNING(log, "HardwareCodecDeflate::doCompressData acquireJob fail!");
+        return 0;
+    }
+    qpl_status status;
+    uint32_t compressed_size = 0;
+
+    job_ptr->op = qpl_op_compress;
+    job_ptr->next_in_ptr = reinterpret_cast<uint8_t *>(const_cast<char *>(source));
+    job_ptr->next_out_ptr = reinterpret_cast<uint8_t *>(dest);
+    job_ptr->available_in = source_size;
+    job_ptr->level = qpl_default_level;
+    job_ptr->available_out = dest_size;
+    job_ptr->flags = QPL_FLAG_FIRST | QPL_FLAG_DYNAMIC_HUFFMAN | QPL_FLAG_LAST | QPL_FLAG_OMIT_VERIFY;
+    status = qpl_execute_job(job_ptr);
+    if (QPL_STS_OK == status)
+    {
+        compressed_size = job_ptr->total_out;
+    }
+    else
+        LOG_WARNING(log, "HardwareCodecDeflate::doCompressData fail ->status: '{}' ", static_cast<size_t>(status));
+
+    DeflateJobHWPool::instance().releaseJob(job_id);
+    return compressed_size;
+}
+
+uint32_t HardwareCodecDeflate::doDecompressData(const char * source, uint32_t source_size, char * dest, uint32_t uncompressed_size) const
+{
+    uint32_t job_id = 0;
+    qpl_job * job_ptr = DeflateJobHWPool::instance().acquireJob(&job_id);
+    if (job_ptr == nullptr)
+    {
+        LOG_WARNING(log, "HardwareCodecDeflate::doDecompressData acquireJob fail!");
+        return 0;
+    }
+    qpl_status status;
+
+    // Performing a decompression operation
+    job_ptr->op = qpl_op_decompress;
+    job_ptr->next_in_ptr = reinterpret_cast<uint8_t *>(const_cast<char *>(source));
+    job_ptr->next_out_ptr = reinterpret_cast<uint8_t *>(dest);
+    job_ptr->available_in = source_size;
+    job_ptr->available_out = uncompressed_size;
+    job_ptr->flags = QPL_FLAG_FIRST | QPL_FLAG_LAST;
+
+    // Decompression
+    status = qpl_execute_job(job_ptr);
```

Comment:
Here, and in the related methods, it would be nice to use C++17 if initializers:

```
if (qpl_status status = qpl_execute_job(job_ptr); status != QPL_STS_OK)
{
    ...
```

Reply:
Done